### PR TITLE
Options for internal uses

### PR DESCRIPTION
**Pull request by akawashiro**
I implemented two options for internal use; `--exclude-runpath-contains` and `delete-unused-DT_STRTAB`

### CODE REVIEW

File: sold.h [hunk 1]
```diff
@@ -270,9 +274,23 @@ class Sold {
...
+            std::vector<char> buf(phdr->p_filesz, 0);
+            std::memcpy(buf.data(), bin->head() + phdr->p_offset, phdr->p_filesz);
+            if (delete_unused_DT_STRTAB_) {
+                LOG(INFO) << SOLD_LOG_BITS(phdr->p_offset) << SOLD_LOG_BITS(bin->dt_strtab())
+                          << SOLD_LOG_BITS(bin->dt_strtab() + bin->strsz()) << SOLD_LOG_BITS(phdr->p_offset + phdr->p_filesz);
+                if (phdr->p_offset <= bin->dt_strtab() && bin->dt_strtab() + bin->strsz() <= phdr->p_offset + phdr->p_filesz) {
+                    LOG(INFO) << "Delete unused DT_STRTAB: " << SOLD_LOG_KEY(bin->dt_strtab() - phdr->p_offset)
+                              << SOLD_LOG_KEY(bin->dt_strtab() - phdr->p_offset + bin->strsz()) << SOLD_LOG_KEY(bin->strsz());
+                    for (int i = bin->dt_strtab() - phdr->p_offset; i < bin->dt_strtab() - phdr->p_offset + bin->strsz(); i++) {
+                        buf[i] = 'Y';
```
Comment (shinh) [inline sold.h:289]: Better to use `'\0'`?

File: sold.h [hunk 2]
```diff
@@ -270,9 +274,23 @@ class Sold {
...
             Elf_Phdr* phdr = load.orig;
             LOG(INFO) << "Emitting code of " << bin->name() << " from " << HexString(ftell(fp)) << " => " << HexString(load.emit.p_offset)
-                      << " + " << HexString(phdr->p_filesz);
+                      << " + " << HexString(phdr->p_filesz) << SOLD_LOG_KEY(delete_unused_DT_STRTAB_);
             EmitPad(fp, load.emit.p_offset);
-            WriteBuf(fp, bin->head() + phdr->p_offset, phdr->p_filesz);
+
+            std::vector<char> buf(phdr->p_filesz, 0);
+            std::memcpy(buf.data(), bin->head() + phdr->p_offset, phdr->p_filesz);
+            if (delete_unused_DT_STRTAB_) {
```
Comment (shinh) [inline sold.h:282]: This just a comment for future change, not a review request: Ideally, it'd be better if we can discard information of all unused data (e.g., DT_SYMTAB, DT_GNU_HASH, ...). To implement it, it is probably to better to populate list of unnecessary regions first and then fill zeros for these regions.

**Review by shinh — APPROVED** [review]
LGTM!

File: sold.h (hunk 1 above)
Comment (akawashiro) [reply sold.h]: Done in https://github.com/akawashiro/sold/pull/10/commits/70b42657f25f015f5440eca499ca9f2ad5597a7b.

File: sold.h (hunk 2 above)
Comment (akawashiro) [reply sold.h:282]: I agree. It sounds good.

**Pull request merged**